backend/src/main/java/com/soen341/instagram/service/impl/RegistrationService.java

```diff
@@ -0,0 +1,76 @@
+package com.soen341.instagram.service.impl;
+
+import java.util.Date;
+import java.util.regex.Pattern;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import com.soen341.instagram.dao.impl.AccountRepository;
+import com.soen341.instagram.exception.account.EmailTakenException;
+import com.soen341.instagram.exception.account.InvalidEmailFormatException;
+import com.soen341.instagram.exception.account.InvalidUsernameFormatException;
+import com.soen341.instagram.exception.account.UsernameTakenException;
+import com.soen341.instagram.model.Account;
+
+@Service("registrationService")
+public class RegistrationService
+{
+	@Autowired
+	private AccountRepository accountRepository;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	public void createNewAccount(final Account account)
+	{
+		if (!isEmailTaken(account.getEmail()) && !isUsernameTaken(account.getUsername()))
+		{
+			if (!isEmailFormatValid(account.getEmail()))
+			{
+				throw new InvalidEmailFormatException();
+			}
+
+			if (!isUsernameFormatValid(account.getUsername()))
+			{
+				throw new InvalidUsernameFormatException();
+			}
+
+			account.setPassword(passwordEncoder.encode(account.getPassword()));
+			account.setCreated(new Date());
+			accountRepository.save(account);
+		}
+	}
+
+	protected boolean isEmailTaken(final String email)
```

Comment:
Should perhaps make these helper methods private. Also, I think it would be more proper to not throw the exception from here. Currently, `isEmailTaken` and `isUsernameTaken` can never return true so there is no need for a return value.
You should either: 
1- Rename the methods to something like `checkIfEmailTaken` and `checkIfUsernameTaken`, make them return nothing, and do nothing except throw an exception if the email/username already exists.
2- Keep them as `isEmailTaken`, `isUsernameTaken`, keep their return value as boolean, but throw the exceptions in the calling method (i.e. createNewAccount)

Reply:
Agreed, going for option 1.